modules/Individual Needs/investigations_manage_addProcess.php

```diff
@@ -0,0 +1,86 @@
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+use Gibbon\Comms\NotificationSender;
+use Gibbon\Domain\System\NotificationGateway;
+use Gibbon\Domain\IndividualNeeds\INInvestigationGateway;
+use Gibbon\Services\Format;
+
+require_once '../../gibbon.php';
+
+$gibbonPersonID = $_GET['gibbonPersonID'] ?? '';
+$gibbonRollGroupID = $_GET['gibbonRollGroupID'] ?? '';
+$gibbonYearGroupID = $_GET['gibbonYearGroupID'] ?? '';
+
+$URL = $_SESSION[$guid]['absoluteURL']."/index.php?q=/modules/Individual Needs/investigations_manage_add.php&gibbonPersonID=$gibbonPersonID&gibbonRollGroupID=$gibbonRollGroupID&gibbonYearGroupID=$gibbonYearGroupID";
+
+if (isActionAccessible($guid, $connection2, '/modules/Individual Needs/investigations_manage_add.php') == false) {
+    $URL .= '&return=error0';
+    header("Location: {$URL}");
+    exit;
+} else {
+    // Proceed!
+    $investigationGateway = $container->get(INInvestigationGateway::class);
+
+    $data = [
+        'gibbonSchoolYearID'    => $gibbon->session->get('gibbonSchoolYearID'),
+        'gibbonPersonIDCreator' => $gibbon->session->get('gibbonPersonID'),
+        'gibbonPersonIDStudent' => $_POST['gibbonPersonIDStudent'] ?? '',
+        'status'                => 'Referral',
+        'date'                  => Format::dateConvert($_POST['date']) ?? '',
+        'reason'                => $_POST['reason'] ?? '',
+        'strategiesTried'       => $_POST['strategiesTried'] ?? '',
+        'parentsInformed'       => $_POST['parentsInformed'] ?? '',
+        'parentsResponse'       => $_POST['parentsResponse'] ?? null
+    ];
+
+    // Validate the required values are present
+    if (empty($data['gibbonPersonIDStudent']) || empty($data['date']) || empty($data['reason']) || empty($data['parentsInformed'])) {
+        $URL .= '&return=error1';
+        header("Location: {$URL}");
+        exit;
+    }
+
+    // Create the record
+    $gibbonINInvestigationID = $investigationGateway->insert($data);
+
+    //Notify form tutors
+    $notificationGateway = new NotificationGateway($pdo);
+    $notificationSender = new NotificationSender($notificationGateway, $gibbon->session);
+
+    $criteria = $investigationGateway->newQueryCriteria();
+    $investigation = $investigationGateway->queryInvestigationsByID($criteria, $gibbonINInvestigationID, $_SESSION[$guid]['gibbonSchoolYearID']);
```

Comment:
The built-in `getByID()` method may be useful here. 

```suggestion
    $investigation = $investigationGateway->getByID($gibbonINInvestigationID);
```

Reply:
I thought about using getByID, but there are a bunch of joined tables, and so I thought it best to use a custom-built method. Have I missed something in how getByID works?

Reply:
Oh, I see. In this case, rather than a query (which is intended for multiple records in DataTables), a `getInvestigationByID` method may be handy. I'm happy to push an example once it;s merged, feel free to discard the other similar suggestions in this case.